.github/unittest/linux_libs/scripts_pettingzoo/environment.yml

```diff
@@ -20,4 +20,4 @@ dependencies:
     - expecttest
     - pyyaml
     - autorom[accept-rom-license]
-    - pettingzoo[all]==1.24.1
+    - pettingzoo[all]==1.24.3
```

Comment:
We are bounding the version in the tests since we do not want future releases of the lib to cause failure in our tests (or to inject code).

However, I would not prevent using the wrapper with future versions of the lib by default as they may be compatible.

Reply:
> However, I would not prevent using the wrapper with future versions of the lib by default as they may be compatible

either we guarantee forward compatibility or we don't but either way we need to relax the tests or enforce the version. Otherwise we risk having errors being reported because of bc-breaking changes on their side

Reply:
Relaxing the version will open them to bc-breaking changes on their side and will obfuscate if the source of the failing tests is us or the lib.

Furthermore, I cannot guarantee I will be available to fix any failing CI error due to bc-compatibility on their side in the future.

Preventing to use the wrapper with some version is too constraining. 

My preferred solution is to test a specific version and say that using other versions is allowed but not tested.

Otherwise, I would remove the version in the tests, but I cannot guarantee I will be the one long term updating our side of PettingZoo

Reply:
> Furthermore, I cannot guarantee I will be available to fix any failing CI error due to bc-compatibility on their side in the future.

Same as above, if we can't commit to solve upcoming issues we should not allow more recent versions. 
This is like pinning the version of tensordict and torch in the releases, it's good practice. 
If we don't we need to remove the version check in the tests otherwise anyone who does a pip install pettingzoo faces issues when using it with torchrl and we're not helping neither ourselves nor users.

A warning is fine but we should make sure users are made aware of when they're off charts